src/plugins/procmon/procmon.cpp

```diff
@@ -371,6 +386,55 @@ static event_response_t process_create_ex_return_hook(drakvuf_t drakvuf, drakvuf
     return VMI_EVENT_RESPONSE_NONE;
 }
 
+static event_response_t process_create_return_hook(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+    auto plugin = get_trap_plugin<procmon>(info);
+    auto params = get_trap_params<process_create_result_t>(info);
+
+    if (!params->verify_result_call_params(drakvuf, info))
+        return VMI_EVENT_RESPONSE_NONE;
+
+    addr_t process_handle_addr = params->process_handle_addr;
+    reg_t status = info->regs->rax;
+
+    ACCESS_CONTEXT(ctx,
+        .translate_mechanism = VMI_TM_PROCESS_DTB,
+        .dtb = info->regs->cr3,
+        .addr = process_handle_addr,
+    );
+
+    addr_t process_handle;
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+    if (VMI_FAILURE == vmi_read_addr(vmi, &ctx, &process_handle))
+        process_handle = 0;
+
+    drakvuf_release_vmi(drakvuf);
+
+    vmi_pid_t new_pid;
+    if (!drakvuf_get_pid_from_handle(drakvuf, info, process_handle, &new_pid))
+        new_pid = 0;
+
+    if (strstr(info->attached_proc_data.name,"notepad"))
+      // somehow stop this run
```

Comment:
at the moment just please add something like

```
printf("should block this\n");
```
instead of

```
// somehow stop this run
```

and check if the printout is displayed correctly

Reply:
I tried this. It didn't work with `NtCreateProcess`. Though, similar kind of snippet in `NtCreateUserProcess` worked in https://github.com/manorit2001/drakvuf/pull/1/commits/1bc9acb32e2b1dc949e07d536c09925ac8820e7b